dm/syncer/syncer.go

```diff
@@ -3355,45 +3363,46 @@ func (s *Syncer) Close() {
 	if s.isClosed() {
 		return
 	}
-
-	s.stopSync()
+	s.stopStreamAndRollbackCheckpoint()
 	s.closeDBs()
-
 	s.checkpoint.Close()
-
 	if err := s.schemaTracker.Close(); err != nil {
 		s.tctx.L().Error("fail to close schema tracker", log.ShortError(err))
 	}
-
 	if s.sgk != nil {
 		s.sgk.Close()
 	}
-
 	s.closeOnlineDDL()
-
 	// when closing syncer by `stop-task`, remove active relay log from hub
 	s.removeActiveRelayLog()
-
 	metrics.RemoveLabelValuesWithTaskInMetrics(s.cfg.Name)
-
 	s.closed.Store(true)
 }
 
-// stopSync stops syncing, now it used by Close and Pause
-// maybe we can refine the workflow more clear.
-func (s *Syncer) stopSync() {
-	if s.done != nil {
-		<-s.done // wait Run to return
-	}
-	s.closeJobChans()
-	s.wg.Wait() // wait job workers to return
+// Kill kill syncer without graceful.
+func (s *Syncer) Kill() {
+	s.tctx.L().Warn("kill syncer without graceful")
+	s.runCancel()
+	s.syncCancel()
+	s.Close()
+}
 
+// stopStreamAndRollbackCheckpoint stop stream and rollback checkpoint now it used by Close() and Pause().
```

Comment:
```suggestion
// stopStreamAndRollbackCheckpoint stops stream and rollbacks checkpoint, now it is used by Close() and Pause().
```

I think we do not need to rollback checkpoint when closing, and it's not intuitive that group close streamer and checkpoint into one function.

Reply:
renamd back to `stopSync` now, i just follow the old logic : rollback checkpoint before syncer.Process return in [here](https://github.com/pingcap/tiflow/blob/9d7ae48d3fee0ea02bfa9dd504e06166bc25b45e/dm/syncer/syncer.go#L703)

Reply:
I think we can only rollback schema checkpoint and tracker when `Pause`, not when `Close`. PTAL @GMHDBJD

Reply:
ping @GMHDBJD  please help me clean the logic about the right time to rollback checkpoint

Reply:
Yes, we rollback when task paused. both pause-task and ***pause by error***